### PR TITLE
WIP: Write info about assets kept during cleanup to db

### PR DESCRIPTION
**Pull request by Martchus**
See https://progress.opensuse.org/issues/16496

Note that this is completely untested and the actual database migration is not contained yet. I just want some feedback whether this approach could work.

---
**Actual commit description:**

* Store the job which prevents the cleanup in the asset table
* The relevant job group and the date of the last use can be
  obtained from that job

Note that there could be multiple jobs/groups preventing an
asset from being deleted. So far only the last one is recorded.

Further steps:
* Show the job group and date of last use in the assets table
* Accumulate the size of the assets kept during cleanup for each
  job group and display it somewhere.

### CODE REVIEW

File: lib/OpenQA/Schema/Result/Assets.pm [hunk 1]
```diff
@@ -242,52 +243,108 @@ sub limit_assets {
...
     for my $id (keys %keep) {
-        OpenQA::Utils::log_debug("KEEP $toremove{$id} $keep{$id}") if $debug_keep;
+        my $asset = $keep{$id};
+
+        OpenQA::Utils::log_debug("KEEP $toremove{$id} $asset->{log_msg}") if $debug_keep;
         delete $toremove{$id};
+
+        # set the time of the last use and the related job group
+        my $job = $jobs->find($asset->{job_id});
+        $asset_resultset->find($id)->update(
```
Comment (coolo) [inline lib/OpenQA/Schema/Result/Assets.pm]: I assume this find($id)->update happens in one SQL?  and why do you query $job here?

Comment (Martchus) [reply lib/OpenQA/Schema/Result/Assets.pm]: In the first place I wanted to store the date of the last use directly in the asset table. This is a leftover, no reason for the query anymore.

**Comment by Martchus** [comment]
Writing last jobs to the database and displaying them in the table basically works. So at least you can now see the job and its group in the table. Clicking on the group leads to the page where you can adjust the size limit for that group. I also extended the test suite.

screenshot_20171128_165259

The last point "Accumulate the size of the assets kept during cleanup for each
job group and display it somewhere." is still TODO.

I'm also thinking where to put that figure. On the job group settings beside the size limit would make sense. However, a sortable list with all job groups and their asset disk usage might be more useful?

**Comment by foursixnine** [comment]
I think the list with jobgroups and asset disk usage makes more sense.

**Comment by coolo** [comment]
how it's rendered is secondary to this PR - I want the info in the database, so we can pull it of the production DB to play with rendering of the massive data.

**Comment by coolo** [comment]
actually these schema conflicts get on my nerves enough to consider redoing the deployment handler myself - there just aren't good options available :(

**Comment by coolo** [comment]
superseded by #1524

**Pull request closed without merging**